serve/mlc_serve/model/tvm_model.py

```diff
@@ -270,18 +295,27 @@ def generate_multi_query(
         torch.cuda.nvtx.range_pop()
 
         last_query_logits = torch.from_dlpack(logits)[last_query_offsets]
-
         return sample_from_logits(
-            last_query_logits, sequence_ids, requests, self.vocab_size
+            last_query_logits,
+            sequence_ids,
+            requests,  # type: ignore
+            sampling_metadata,
+            self.vocab_size,
+            self._copy_stream,
+            self.torch_dtype,
+            self.torch_dev,
+            past_decode_tokens,
         )
 
     def generate(
         self,
-        requests: RequestsType,
+        requests: List[RequestType],
         cache: KVCacheInfo,
     ) -> List[TextGenerationResult]:
-        if len(requests) == 0:
+        batch_size = len(requests)
+        if batch_size == 0:
             return []
+        LOG.debug(f"Batch size: {batch_size}")
```

Comment:
Move above the `if` so we always know what the batch size, and improve the message. 

In struct log you should always do 
```suggestion
        LOG.debug(f"Generation batch size: f{batch_size}.", batch_size=batch_size)
```

By passing as keyword arguments it becomes inspectable as an integer in the logs.